samples/nrf9160/custom_trace_medium/src/main.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <nrf_modem_at.h>
+#include <modem/lte_lc.h>
+#include <modem/lte_lc.h>
+#include <modem/nrf_modem_lib.h>
+
+/* Network registration semaphore */
+static K_SEM_DEFINE(cfun_sem, 0, 1);
+
+/* define callback */
+LTE_LC_ON_CFUN(cfun_hook, on_cfun, NULL);
+
+/* callback implementation */
+static void on_cfun(enum lte_lc_func_mode mode, void *context)
+{
+	printk("LTE mode changed to %d\n", mode);
+}
+
+void main(void)
+{
+	int err;
+
+	printk("Custom trace medium sample started\n");
+
+	printk("Connecting to network\n");
+	err = lte_lc_func_mode_set(LTE_LC_FUNC_MODE_NORMAL);
+	if (err) {
+		printk("Failed to start modem\n");
+		return;
+	}
+
```

Comment:
We should use the `cfun_sem` here to wait for the device to be connected before shutting down (or time out after some time).